codegen-luau/runtime/runtime.lua

```diff
@@ -96,8 +116,8 @@ do
 	function nearest.num(num)
 		local result = math_round(num)
 
-		if math_abs(num) % 1 == 0.5 and temp_2 % 2 == 1 then
-			result = result - 1
+		if math_abs(num) % 1 == 0.5 and math_floor(math_abs(num) % 2) == 0 then
+			result -= math_sign(result)
```

Comment:
I'd use `result = result - math_sign(result)` still as the plan is to minimize Luau-specific syntax because of tooling that does not yet support it.